### ISSUE TITLE
lockfiles .gitignored

### ISSUE THREAD

**Issue by kandros** [apollographql/react-apollo#1585]
considering how easy it is for a single package dependency version missmatch to break the whole project, why are the lockfiles non in version control?

**Comment by jaydenseric** [comment]
See https://github.com/sindresorhus/ama/issues/479#issuecomment-310661514.

**Comment by magicmaaaaan** [comment]
@jaydenseric That ideology comes from [Ruby's bundler](http://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/). [yarn's argument](https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/) (also linked in that issue) makes more sense for Javascript projects, i.e., commit the lockfile regardless if it's an app or lib.

**Comment by jaydenseric** [comment]
@magicmaaaaan See https://github.com/graphcool/graphql-config/pull/80#issuecomment-359971698.

**Comment by kandros** [comment]
TLDR;  wanted to contribute, had to spend hours to find and fix a depency of dependy (in beta) change.

Without lock files the dev experience is likely to be non deterministic

**Comment by hwillson** [comment]
This should no longer be an issue with current versions of React Apollo. Thanks!
